packages/base/src/AbstractProvider.ts

```diff
@@ -4,28 +4,22 @@ import {
   CellDep,
   ChainInfo,
   Hash,
-  Hexadecimal,
   HexNumber,
   HexString,
   Input,
   Script,
   Transaction,
   TxPoolInfo,
 } from '@ckb-lumos/base';
-import { Config as LumosConfig, predefined, ScriptConfig } from '@ckb-lumos/config-manager';
+import { predefined, ScriptConfig } from '@ckb-lumos/config-manager';
 import { encodeToAddress, generateAddress, parseAddress } from '@ckb-lumos/helpers';
 import { generateTypeIdScript } from './typeid';
-import { Provider, ResolvedOutpoint } from './';
-
-type OptionalConfig = {
-  MIN_FEE_RATE: Hexadecimal;
-};
-export type ProviderConfig = LumosConfig & OptionalConfig;
-export type InitOptions<T extends LumosConfig = LumosConfig> = Omit<T, keyof OptionalConfig> & Partial<OptionalConfig>;
+import { Provider, ProviderConfig, InitOptions, ResolvedOutpoint, CellOutPointProvider } from './';
 
 export abstract class AbstractProvider implements Provider {
   private initialized = false;
   private _config: ProviderConfig | undefined;
+  abstract depOutPointProvider: CellOutPointProvider | undefined;
```

Comment:
maybe we dont need depOutPointProvider here, override `getCellDep` is fine

Reply:
Done, good advice, didn't get used to overriding before.